de.metas.business/src/main/java-legacy/org/compiere/model/CalloutInOut.java

```diff
@@ -235,115 +238,81 @@ else if (X_C_DocType.DOCBASETYPE_MaterialReceipt.equals(DocBaseType))
-				else { // get default PriceList
-					int i = Env.getContextAsInt(ctx, "#M_PriceList_ID");
-					if (i != 0)
-						mTab.setValue("M_PriceList_ID", new Integer(i));
-				}
-				int shipTo_ID = rs.getInt("C_BPartner_Location_ID");
-				// overwritten by InfoBP selection - works only if InfoWindow
-				// was used otherwise creates error (uses last value, may belong
-				// to differnt BP)
-				if (C_BPartner_ID.toString().equals(
-						Env.getContext(ctx, WindowNo, Env.TAB_INFO,
-								"C_BPartner_ID"))) {
-					String loc = Env.getContext(ctx, WindowNo, Env.TAB_INFO,
-							"C_BPartner_Location_ID");
-					if (loc.length() > 0)
-						shipTo_ID = Integer.parseInt(loc);
-				}
-				if (shipTo_ID == 0)
-					mTab.setValue("C_BPartner_Location_ID", null);
-				else
-					mTab.setValue("C_BPartner_Location_ID", new Integer(
-							shipTo_ID));
-				//metas: B062 ende
-				
-				if (!IsSOTrx) {
-					// Location
-					ii = new Integer(rs
-							.getInt("C_BPartner_Location_ID"));
-					if (rs.wasNull())
-						mTab.setValue("C_BPartner_Location_ID", null);
-					else
-						mTab.setValue("C_BPartner_Location_ID", ii);
-					// Contact
-					ii = new Integer(rs.getInt("AD_User_ID"));
-					if (rs.wasNull())
-						mTab.setValue("AD_User_ID", null);
-					else
-						mTab.setValue("AD_User_ID", ii);
-				}
+		//
+		// BPartner Location (i.e. ShipTo)
+		final I_C_BPartner_Location shipToLocation = suggestShipToLocation(ctx, WindowNo, bpartner);
+		inout.setC_BPartner_Location(shipToLocation);
 
-				// Bugs item #1679818: checking for SOTrx only
-				if (IsSOTrx) {
-					// CreditAvailable
-					double CreditAvailable = rs.getDouble("CreditAvailable");
-					if (!rs.wasNull() && CreditAvailable < 0)
-						mTab.fireDataStatusEEvent("CreditLimitOver",
-								DisplayType.getNumberFormat(DisplayType.Amount)
-										.format(CreditAvailable), false);
-				}//
+		//
+		// BPartner Contact
+		if (!isSOTrx)
+		{
+			I_AD_User contact = null;
+			if (shipToLocation != null)
+			{
+				contact = Services.get(IBPartnerBL.class).retrieveUserForLoc(shipToLocation);
 			}
-		} catch (SQLException e) {
-			log.error(sql, e);
-			return e.getLocalizedMessage();
-		} finally {
-			DB.close(rs, pstmt);
+			if (contact == null)
+			{
+				contact = Services.get(IBPartnerBL.class).retrieveShipContact(bpartner);
+			}
+			inout.setAD_User(contact);
 		}
 
-		return "";
+		//
+		// Check SO credit available
+		if (isSOTrx)
+		{
+			final IBPartnerStats bpartnerStats = Services.get(IBPartnerStatsBL.class).getBPartnerStats(bpartner);
+			final BigDecimal soCreditUsed = bpartnerStats.getSoCreditUsed();
+			if (soCreditUsed.signum() < 0)
+			{
+				mTab.fireDataStatusEEvent("CreditLimitOver", DisplayType.getNumberFormat(DisplayType.Amount).format(soCreditUsed), false);
+			}
+		}
+		return NO_ERROR;
 	} // bpartner
+	
+	public static I_C_BPartner_Location suggestShipToLocation(final Properties ctx, final int windowNo, final I_C_BPartner bpartner)
```

Comment:
Yes.... and basically if u look at that method, you will see that is callout specific (i.e. it's checking the window context for C_BPartner_ID/C_BPartner_Location_ID that were set by Info BPartner).